admittance_controller/include/admittance_controller/admittance_rule_impl.hpp

```diff
@@ -280,16 +272,13 @@ bool AdmittanceRule::calculate_admittance_rule(AdmittanceState & admittance_stat
   auto F_base = admittance_state.wrench_base;
 
   // zero out any forces in the control frame
-  Eigen::Matrix<double, 6, 1> F_control;
-  F_control.block<3, 1>(0, 0) = rot_base_control.transpose() * F_base.block<3, 1>(0, 0);
-  F_control.block<3, 1>(3, 0) = rot_base_control.transpose() * F_base.block<3, 1>(3, 0);
+  Eigen::Matrix<double, 6, 1> F_control = rot_base_control_6d.transpose() * F_base;
   F_control = F_control.cwiseProduct(admittance_state.selected_axes);
-  F_base.block<3, 1>(0, 0) = rot_base_control * F_control.block<3, 1>(0, 0);
-  F_base.block<3, 1>(3, 0) = rot_base_control * F_control.block<3, 1>(3, 0);
 
   // Compute admittance control law in the base frame: F = M*x_ddot + D*x_dot + K*x
-  Eigen::Matrix<double, 6, 1> X_ddot =
-    admittance_state.mass_inv.cwiseProduct(F_base - D * X_dot - K * X);
+  Eigen::Matrix<double, 6, 1> X_ddot = rot_base_control_6d * M_inv *
+                                       (F_control - D * rot_base_control_6d.transpose() * X_dot -
+                                        K * rot_base_control_6d.transpose() * X);
```

Comment:
This looks good, but it might be a good idea to update the comment "F = M*x_ddot + D*x_dot + K*x"

Reply:
I added my proposal to finish this.

Reply:
@pac48 could you give this a quick review please?